static/app/views/performance/transactionSummary/transactionEvents/eventsTable.tsx

```diff
@@ -322,7 +322,13 @@ class EventsTable extends Component<Props, State> {
         >
           {({pageLinks, isLoading, tableData}) => {
             const parsedPageLinks = parseLinkHeader(pageLinks);
-            const currentEvent = parsedPageLinks?.next?.cursor.split(':')[1] ?? 0;
+            let currentEvent = parsedPageLinks?.next?.cursor.split(':')[1] ?? 0;
+            if (
+              parseInt(currentEvent.replace(',', ''), 10) >
+              parseInt(totalEventCount.replace(',', ''), 10)
+            ) {
+              currentEvent = totalEventCount;
+            }
```

Comment:
Total nit, but this might be even clearer with more specific variable names:
```suggestion
            const parseHumanizedNumber = (number: string) => parseInt(number.replace(',', ''), 10);
            
            const cursorSize = parseHumanizedNumber(parsedPageLinks?.next?.cursor.split(':')[1] ?? 0);
            const eventCount = parseHumanizedNumber(totalEventCount);
            
            const visibleEventCount = Math.min(cursorSize, eventCount);
```

Feel free to ignore, I was just surprised by the mechanism here (the total size is a string, and the cursor size must be parsed from headers!)

Reply:
I do like this better, it is more readable, and yes, I agree there definitely can be a better mechanism to get the total size! Something we can look into refactoring